### PR TITLE
Remove intra-doc links

### PR DESCRIPTION
**Pull request by kleimkuhler**
Any documentation links that were `crate` or path dependent have been updated
to use the relative paths of the generated Rustdoc output.

Signed-off-by: Kevin Leimkuhler <kleimkuhler@icloud.com>

### CODE REVIEW

File: src/poll.rs [hunk 1]
```diff
@@ -413,17 +408,22 @@ impl Registry {
     ///
     /// # Notes
     ///
+    /// Callers must ensure that if a source being registered with a `Poll`
```
Comment (Thomasdezeeuw) [inline src/poll.rs]: 👍

File: src/poll.rs [hunk 2]
```diff
@@ -184,16 +184,11 @@ use std::{fmt, io};
 /// data to be copied into an intermediate buffer before it is passed to the
 /// kernel.
 ///
-/// Notifications generated by [`SetReadiness`] are handled by an internal
```
Comment (Thomasdezeeuw) [inline src/poll.rs]: 👍

**Comment by hawkw** [comment]
Hmm, my understanding is that intra-doc links should be _preferred_ since relative links can sometimes break due to path differences between docs.rs and local docs — @carllerche may remember the specifics?

File: src/lib.rs [hunk 3]
```diff
@@ -17,8 +17,8 @@
 //!
 //! For more detail, see [`Poll`].
 //!
-//! [`Poll`]: struct.Poll.html
-//! [`Events`]: struct.Events.html
+//! [`Poll`]: ../mio/struct.Poll.html
```
Comment (Thomasdezeeuw) [inline src/lib.rs]: `../mio` can be replaced with `./`.

File: src/lib.rs [hunk 4]
```diff
@@ -30,7 +30,7 @@
 //! ## Guide
 //!
 //! A getting started guide is available in the
-#![cfg_attr(feature = "guide", doc = "[`guide`](crate::guide) module.")]
+#![cfg_attr(feature = "guide", doc = "[`guide`](../mio/guide/index.html) module.")]
```
Comment (Thomasdezeeuw) [inline src/lib.rs]: `../mio` can be replaced with `./`.

**Comment by Thomasdezeeuw** [comment]
@kleimkuhler thanks for doing this, last issue for the alpha1 milestone.

**Comment by kleimkuhler** [comment]
@hawkw Intra-doc links are a nightly feature. `cargo doc --open` on master just had broken links in all those places. Only if a user knew to use `cargo +nightly doc --open` did they work properly.

**Comment by kleimkuhler** [comment]
The same issues exists on Tokio master as well right now. If docs are not compiled locally with nightly then most of the links do not work 😕

**Comment by Thomasdezeeuw** [comment]
And to add in the fun the nightly compiler doesn't give any warnings that you're using a nightly only feature.

**Pull request merged**